ahnlich/types/src/ai/query.rs

```diff
@@ -1 +1,53 @@
+use super::{AIModel, AIStoreType};
+use crate::keyval::{StoreKey, StoreName, StoreValue};
+use crate::metadata::{MetadataKey, MetadataValue};
+use crate::predicate::PredicateCondition;
+use crate::similarity::Algorithm;
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use std::num::NonZeroUsize;
 
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub enum AiQuery {
+    CreateStore {
+        r#type: AIStoreType,
+        store: StoreName,
+        model: AIModel,
+        predicates: HashSet<MetadataKey>,
+    },
+    GetPred {
+        store: StoreName,
+        condition: PredicateCondition,
+    },
+    GetSimN {
+        store: StoreName,
+        search_input: MetadataValue,
```

Comment:
Realizing this might be a bit confusing to users, we can have a From conversion that we use to convert to MetadataValue, but create a separate type maybe called StoreInput with the same RawString and Binary variants